src/lib/operators/jit_operator/operators/jit_write_reference.cpp

```diff
@@ -0,0 +1,107 @@
+#include "jit_write_reference.hpp"
+
+#include "../jit_types.hpp"
+#include "constant_mappings.hpp"
+#include "resolve_type.hpp"
+#include "storage/base_value_segment.hpp"
+#include "storage/value_segment.hpp"
+
+namespace opossum {
+
+std::string JitWriteReference::description() const {
+  std::stringstream desc;
+  desc << "[WriteReference] ";
+  for (const auto& output_column : _output_columns) {
+    desc << output_column.column_name << " = Col#" << output_column.referenced_column_id << ", ";
+  }
+  return desc.str();
+}
+
+std::shared_ptr<Table> JitWriteReference::create_output_table(const Table& in_table) const {
+  TableColumnDefinitions column_definitions;
+
+  const auto& input_column_definitions = in_table.column_definitions();
+
+  for (const auto& output_column : _output_columns) {
+    auto column_definition = input_column_definitions[output_column.referenced_column_id];
+    column_definition.name = output_column.column_name;
+    column_definitions.push_back(column_definition);
+  }
+
+  return std::make_shared<Table>(column_definitions, TableType::References);
+}
+
+void JitWriteReference::before_query(Table& out_table, JitRuntimeContext& context) const {
+  context.output_pos_list = std::make_shared<PosList>();
+}
+
+void JitWriteReference::after_chunk(const std::shared_ptr<const Table>& in_table, Table& out_table,
+                                    JitRuntimeContext& context) const {
+  if (context.output_pos_list->size() > 0) {
+    Segments out_segments;
+
+    out_segments.reserve(_output_columns.size());
+
+    const auto chunk_in = in_table->get_chunk(context.chunk_id);
+    if (in_table->type() == TableType::References) {
+      auto filtered_pos_lists = std::map<std::shared_ptr<const PosList>, std::shared_ptr<PosList>>{};
+
+      for (const auto& output_column : _output_columns) {
```

Comment:
The order in which methods are called seems to be important, but is not documented / asserted.